packages/server/src/oauth/token.ts

```diff
@@ -352,8 +352,10 @@ export async function exchangeExternalAuthToken(
   let userInfo;
   try {
     userInfo = await getExternalUserInfo(idp, subjectToken);
-  } catch (err) {
-    sendTokenError(res, 'invalid_request', normalizeErrorString(err));
+  } catch (err: any) {
+    // If the status is 429, then the error message will be Too Many Requests
+    const status = err.message === 'Too Many Requests' ? 429 : 400;
```

Comment:
This is kinda brittle.  We have some helper utilities that might be good here:

* `function normalizeOperationOutcome(error: unknown): OperationOutcome`
* `function operationOutcomeToString(outcome: OperationOutcome): string`
* `function getStatus(outcome: OperationOutcome): number`

```ts
} catch (err) {
    const outcome = normalizeOperationOutcome(err);
    sendTokenError(res, 'invalid_request', operationOutcomeToString(outcome), getStatus(outcome));
}
```
}